src/main/java/net/minecraftforge/client/model/generators/MultiPartBlockStateBuilder.java

```diff
@@ -127,18 +145,167 @@ JsonObject toJson() {
+            public final List<ConditionGroup> nestedConditionGroups = new ArrayList<>();
+
+            /**
+             * Set a condition for this part, which consists of a property and a set of
+             * valid values. Can be called multiple times for multiple different properties.
+             *
+             * @param <T>    the type of the property value
+             * @param prop   the property
+             * @param values a set of valid values
+             * @return this builder
+             * @throws NullPointerException     if {@code prop} is {@code null}
+             * @throws NullPointerException     if {@code values} is {@code null}
+             * @throws IllegalArgumentException if {@code values} is empty
+             * @throws IllegalArgumentException if {@code prop} has already been configured
+             * @throws IllegalArgumentException if {@code prop} is not applicable to the
+             *                                  current block's state
+             * @throws IllegalStateException    if {@code nestedConditionGroups.size() != 0}
+             */
+            @SafeVarargs
+            public final <T extends Comparable<T>> ConditionGroup condition(Property<T> prop, T... values)
+            {
+                Preconditions.checkNotNull(prop, "Property must not be null");
+                Preconditions.checkNotNull(values, "Value list must not be null");
+                Preconditions.checkArgument(values.length > 0, "Value list must not be empty");
+                Preconditions.checkArgument(!conditions.containsKey(prop), "Cannot set condition for property \"%s\" more than once", prop.getName());
+                Preconditions.checkArgument(canApplyTo(owner), "IProperty %s is not valid for the block %s", prop, owner);
+                Preconditions.checkState(nestedConditionGroups.size() == 0, "Can't have normal conditions if there are already nested condition groups");
+                this.conditions.putAll(prop, Arrays.asList(values));
+                return this;
+            }
+
+            /**
+             * Allows having nested groups of conditions if the current condition group is empty.
+             * @throws IllegalStateException if there are any normal conditions in the current condition group
+             */
+            public final ConditionGroup nestedGroup()
+            {
+                Preconditions.checkState(conditions.size() == 0, "Can't have nested condition groups if there are already normal conditions");
+                ConditionGroup group = new ConditionGroup();
+                group.parent = this;
+                this.nestedConditionGroups.add(group);
+                return group;
+            }
+
+            public IConditionGroup endGroup() 
+            {
+                if (parent == null)
+                {
+                    return MultiPartBlockStateBuilder.PartBuilder.this; 
+                }
+                else {
```

Comment:
This bracket should be on the next line, though it may just be cleaner as they are single line things for this entire thing to be:
```java
return parent == null ? MultiPartBlockStateBuilder.PartBuilder.this : parent;
```